views/js/ResultServerApi.js

```diff
@@ -22,11 +26,26 @@ define(['jquery', 'iframeNotifier'], function($, iframeNotifier) {
     ResultServerApi.prototype.submitItemVariables = function(itemId, serviceCallId, responses, scores, events, params, callback) {
 
         var that = this;
-        var error = function error(){
+        var error = function error(xhr){
+            var response;
+            var message = __('An error occurs, please contact your administrator');
+
+            if (xhr) {
+                try { response = JSON.parse(xhr.responseText); } catch (e) {}
+            }
+
             //there is no error management, so doing an alert (an eval and I'll burn in hell...)
             //TODO manage errors during the delivery
-            alert('An error occurs, please contact your administrator');
-
+            if (response && response.message) {
+                if (response.code === TEST_STATE_CLOSED || response.code === TEST_STATE_SUSPENDED) {
+                    message = false;
+                } else {
+                    message = response.message;
+                }
+            }
+            if (message) {
+                alert(message);
```

Comment:
Sure, it would be fine. Unfortunately I did not succeed to use this component in this context. The browser doesn't display it. I think it is because the stylesheets are not loaded in the frame. However I didn't spend too much time to find a solution, but it can be a good improvement.